### ISSUE TITLE
Confirm market transactions

### ISSUE THREAD

**Issue by benbierens** [codex-storage/nim-codex#744]
Encountered situation during testing where estimate-gas is successful, but transaction is not. Errors are not available because transactions are not confirmed.
Propose to add a 1-block confirmation to the contract calls.

Example from `sales/states/filling.nim`:
```nim
  await market.fillSlot(data.requestId, data.slotIndex, state.proof, collateral)
```

**Comment by emizzle** [comment]
The change would need to be on https://github.com/codex-storage/nim-codex/blob/916e4834d088c9e1837b0a30e8403095fd8b565e/codex/contracts/market.nim#L134

**Comment by AuHau** [comment]
Will be fixed as part of #805
